src/catchup/ApplyBucketsWork.cpp

```diff
@@ -208,8 +290,27 @@ ApplyBucketsWork::doWork()
 {
     ZoneScoped;
 
-    // Step 1: apply buckets. Step 2: assume state
-    if (!mSpawnedAssumeStateWork)
+    // Step 1: index buckets. Step 2: apply buckets. Step 3: assume state
+    bool isUsingBucketListDB = mApp.getConfig().isUsingBucketListDB();
+    if (isUsingBucketListDB)
+    {
+        if (!mIndexBucketsWork)
+        {
+            mIndexBucketsWork = addWork<IndexBucketsWork>(mBucketsToIndex);
+        }
+
+        // If indexing has not finished or finished and failed, return result
+        // status. If indexing finished and succeeded, move on and spawn assume
+        // state work.
+        auto status = checkChildrenStatus();
```

Comment:
Please don't use `checkChildrenStatus` to determine the state of `mIndexBucketsWork`, as this parent work might have other children, which would cause the function to return unexpected results. To query state of each individual work, use that work's `getState` method.